src/api/client/cache.ts

```diff
@@ -44,7 +44,13 @@ const cache = new InMemoryCache({
             // if offset and limit are not provided.
             const offset = args?.offset ?? 0
             const limit = args?.limit ?? existing?.length
-            return existing?.slice(offset, offset + limit)
+            const sortingASC = args?.orderBy === VideoOrderByInput.CreatedAtAsc
+            return sortingASC
+              ? existing?.slice(offset, offset + limit)
+              : existing
+                  ?.slice()
+                  .reverse()
+                  ?.slice(offset, offset + limit)
```

Comment:
Let's split slicing and sorting to improve readability here. Also I'm not sure if `reverse()` call is enough. There is no guarantee that the data in the cache is already sorted. Maybe let's do a proper sort here

Reply:
Fair added full sorting just in case